### PR TITLE
Feature/point of interest structure upgrade

### PR DESCRIPTION
**Pull request by moorchegue**
Addresses #21.

This is really a work in progress, but I'd like to share it early anyway, since it might help catch some gotchas early on as well.

### CODE REVIEW

File: adapters/spreadsheet_adapter.py [hunk 1]
```diff
@@ -19,12 +19,16 @@ def transform(self, source: List[SpreadsheetRow]) -> List[PointOfInterest]:
         return [self.transform_row(row) for row in source]
 
     def transform_row(self, row: SpreadsheetRow) -> PointOfInterest:
+        rename = {
```
Comment (littlepea) [inline adapters/spreadsheet_adapter.py:22]: Why not use existing `ALIASES` above?

File: config/constants.py [hunk 2]
```diff
@@ -1,5 +1,8 @@
 from typing import Set
 
+
+DEFAULT_CATEGORY = 'Any help'
```
Comment (littlepea) [inline config/constants.py:4]: This has already been added by @idisblueflash so your PR should not have this change, please, merge/rebase the latest `master`

File: models/point_of_interest.py [hunk 3]
```diff
@@ -1,13 +1,24 @@
...
-    lat: str = ""
-    lng: str = ""
-    categories: str = ""
-    organizations: str = ""
-    description: str = ""
+    name: str = ''
+    country: str = ''
+    city: str = ''
+    address: str = ''
+    categories: List[str] = []
```
Comment (littlepea) [inline models/point_of_interest.py:10]: Make sure that these convert correctly into CSV and don't break existing unit tests for spiders

File: scraping/spiders/canada_immigration_services.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+import scrapy
+import hjson
+
+from config.constants import DEFAULT_CATEGORY
+
+
+log = logging.getLogger(__name__)
+
+
```
Comment (littlepea) [inline scraping/spiders/canada_immigration_services.py:12]: Why is this file here? Are you combining 2 PRs in one?

File: tests/test_convert_data.py [hunk 5]
```diff
@@ -79,8 +79,15 @@ def test_spreadsheet_data_conversion():
 
     result = usecase.convert_spreadsheet("some-id")
     csv = FILE_VALUES[0]
-    expected_row = "The Ukrainian House,Poland,Warsaw,\"ul. Zamenhofa 1, 00-153\",52.24734033,20.9964833," \
-                   "Accommodation,Fundacja “Nasz Wybór”,Crisis support center"
+    expected_row = 'The Ukrainian House,Poland,Warsaw,' \
```
Comment (littlepea) [inline tests/test_convert_data.py:82]: Are you using some auto-formatter that converts all quotes to single-quotes?

File: scraping/spiders/canada_immigration_services.py (hunk 4 above)
Comment (moorchegue) [reply scraping/spiders/canada_immigration_services.py:12]: I actually thought that it was merged. Is it okay to merge it?

File: tests/test_convert_data.py (hunk 5 above)
Comment (moorchegue) [reply tests/test_convert_data.py:82]: Yeah, one style of quotes does make it a bit easier for me. Do we have a preference what quotes to use?

File: models/point_of_interest.py (hunk 3 above)
Comment (moorchegue) [reply models/point_of_interest.py:10]: The tests I did fix, but export to CSV might actually work a bit differently. Let me poke around it a bit more…

File: config/constants.py (hunk 2 above)
Comment (moorchegue) [reply config/constants.py:4]: Yeah, bad double rebase…

File: adapters/spreadsheet_adapter.py (hunk 1 above)
Comment (moorchegue) [reply adapters/spreadsheet_adapter.py:22]: You're right, exactly same purpose.

File: tests/test_convert_data.py (hunk 5 above)
Comment (littlepea) [reply tests/test_convert_data.py:82]: I agree that one style is better, I don't have a preference but I was leaning towards double-quotes because is later we'll enable some auto-formatter like `black` it's going to force double-quotes

File: scraping/spiders/canada_immigration_services.py (hunk 4 above)
Comment (littlepea) [reply scraping/spiders/canada_immigration_services.py:12]: Yes, you can merge it

**Comment by littlepea** [comment]
> As I was fixing `moldova_dopomoga` crawler (probably source data changed), I noticed that there's also a map there (now): https://www.google.com/maps/d/embed?mid=1RiyRvxOE8q3Z-eo7e298nLn_I0nKXqq6&ehbc=2E312F&ll=47.152161219376964%2C28.378650750000023&z=8
> 
> It seems to have all the same info, and also coordinates for all the points. Would you like me to update the crawler to scrape that info?

It would be nice if it's easy to do 👍

File: tests/test_convert_data.py (hunk 5 above)
Comment (moorchegue) [reply tests/test_convert_data.py:82]: No problem, I'll go with that from now on then.

File: scraping/spiders/canada_immigration_services.py (hunk 4 above)
Comment (moorchegue) [reply scraping/spiders/canada_immigration_services.py:12]: Cool, thanks!

**Comment by moorchegue** [comment]
Sure, no problem!

**Comment by moorchegue** [comment]
Hm… looks like `poetry install` fails on some package in the CI… Have anybody ran into this? I installed locally into a venv with `pip install -e .` , that worked.

Other than that I think I fixed all outstanding issues. Could you guys take a peek, please?

**Comment by littlepea** [comment]
> Hm… looks like `poetry install` fails on some package in the CI… Have anybody ran into this? I installed locally into a venv with `pip install -e .` , that worked.
> 
> Other than that I think I fixed all outstanding issues. Could you guys take a peek, please?

On `master` branch CI is passing...

@wdunn001 any ideas?

File: models/point_of_interest.py [hunk 6]
```diff
@@ -1,13 +1,24 @@
...
+    organizations: List[str] = []
+    description: str = ''
+    lat: str = ''
+    lng: str = ''
+    phone: str = ''
+    email: str = ''
+    url: str = ''
+    socialmedia: str = ''
+    messenger: str = ''
```
Comment (littlepea) [inline models/point_of_interest.py:19]: While you were working on it they've add two more fields to the schema 😂 

```
messenger ->  FB messenger
telegram  -> telegram
whatsapp -> whatsapp
```

**Comment by wdunn001** [comment]
check tests are working correctly something in your latest PR is breaking tests

File: models/point_of_interest.py [hunk 7]
```diff
@@ -1,13 +1,26 @@
...
+    categories: List[str] = []
+    organizations: List[str] = []
+    description: str = ''
+    lat: str = ''
+    lng: str = ''
+    phone: str = ''
+    email: str = ''
+    url: str = ''
+    socialmedia: str = ''
+    fb_messenger: str = ''
```
Comment (littlepea) [inline models/point_of_interest.py:19]: The name of the fields should still be just `messenger`, "FB messenger" was the meaning

File: scraping/spiders/moldova_dopomoga.py [hunk 8]
```diff
@@ -11,8 +12,69 @@
...
+
+
+def parse_response(response):
+    regexp = re.compile(r'var _pageData = "(.*)";', re.S)
+    json_str, = re.search(regexp, response.text).groups()
+    return json.loads(json_str.replace("\\", ""))
+
+
+# XXX: there must be some that encodes/decodes this kind of structure
+def parse_points(json_xml):
```
Comment (littlepea) [inline scraping/spiders/moldova_dopomoga.py:55]: Maybe add a dedicated unit test for this function so that we can see what kind of JSON you're parsing?

In theory, you could deserialize it into a pydantic model

**Comment by littlepea** [comment]
I'm gonna merge it so that we can have the new schema on master, we can address any remaining issues in a separate PR
